rules/rbe_repo.bzl

```diff
@@ -216,56 +214,70 @@ _VERBOSE = False
-                  " to function properly when output_base is set") % _RBE_AUTOCONF_ROOT)
-
-        # Try to use the default project
-        # This is Bazel black magic, we're traversing the directories in the output_base,
-        # assuming that the bazel_toolchains external repo will exist in the
-        # expected path.
-        project_root = ctx.path(".").dirname.get_child("bazel_toolchains").get_child("rules").get_child("cc-sample-project")
-        if not project_root.exists:
-            fail(("Could not find default autoconf project in %s, please make sure " +
-                  "the bazel-toolchains repo is properly imported in your workspace") % str(project_root))
-        project_root = str(project_root)
-        use_default_project = True
-
     name = ctx.attr.name
-    outputs_tar = ctx.attr.name + "_out.tar"
-
-    # Pull the image using 'docker pull'
     image_name = None
     if ctx.attr.digest:
         image_name = ctx.attr.registry + "/" + ctx.attr.repository + "@" + ctx.attr.digest
     else:
         image_name = ctx.attr.registry + "/" + ctx.attr.repository + ":" + ctx.attr.tag
-    _pull_image(ctx, image_name)
 
-    # If tag is specified instead of digest, resolve it to digest in the
-    # image_name as it will be used later on in the platform targets.
-    if ctx.attr.tag:
-        result = ctx.execute(["docker", "inspect", "--format={{index .RepoDigests 0}}", image_name])
-        _print_exec_results("Resolve image digest", result, fail_on_error = True)
-        image_name = result.stdout.splitlines()[0]
-        print("Image with given tag `%s` is resolved to %s" %
-              (ctx.attr.tag, image_name))
+    # If not using checked-in configs we need to pull the container and resolve its tag to digest
+    # before we create the platform target.
+    if not ctx.attr.config_version:
+        # Perform some safety checks
+        _validate_host(ctx)
+        project_root = ctx.os.environ.get(_AUTOCONF_ROOT, None)
+
+        # TODO (nlopezgi): validate _AUTOCONF_ROOT points to a valid Bazel project
+        use_default_project = False
+        if not project_root:
+            if ctx.attr.output_base:
+                fail(("%s env variable must be set for rbe_autoconfig" +
+                      " to function properly when output_base is set") % _AUTOCONF_ROOT)
+
+            # Try to use the default project
+            # This is Bazel black magic, we're traversing the directories in the output_base,
+            # assuming that the bazel_toolchains external repo will exist in the
+            # expected path.
+            project_root = ctx.path(".").dirname.get_child("bazel_toolchains").get_child("rules").get_child("cc-sample-project")
+            if not project_root.exists:
+                fail(("Could not find default autoconf project in %s, please make sure " +
+                      "the bazel-toolchains repo is properly imported in your workspace") % str(project_root))
```

Comment:
suggest changing "properly" to something more explicit/actionable.

Example: Ensure bazel-toolchains repo is import target is named "bazel_toolchains" and imported *before* the rbe_autoconfig target declaration.

We could also explicitly look for "bazel_toolchains" in native.existing_rules().keys() and give an explicit error if bazel_toolchains wasn't imported. If that check passes and this fails then it's more of an internal error because our assumption about Bazel's black magic may have changed

Reply:
fixed (left todo to use native.existing_rules())